src/dense.jl

```diff
@@ -0,0 +1,193 @@
+# A higher level stepper, defined as a wrapper around another stepper.
+
+#TODO: how about having an DenseStepper <: AbstractWrapper <: AbstractStepper?
+immutable DenseStepper <: AbstractStepper
+    solver::Solver
+end
+
+solve(ode::ExplicitODE,
+      stepper::DenseStepper,
+      options::Options) = Solver(ode,stepper,options)
+
+dense(sol::Solver) = solve(sol.ode, DenseStepper(sol), sol.options)
+
+"""
+
+The state of the dense stepper
+
+- s0, s1: Previous steps, used to produce interpolated output
+- solver_state: The state of the associated solver
+- ytmp: work array
+
+"""
+type DenseState{T,S} <: AbstractState
+    s0::Step{T,S}
+    s1::Step{T,S}
+    last_tout::T
+    first_step
+    solver_state::AbstractState
+    # used for storing the interpolation result
+    ytmp::S
+    solver_done
+end
+
+
+function start(s::Solver{DenseStepper})
+    # extract the real solver
+    solver = s.stepper.solver
+    t0  = solver.ode.t0
+    y0  = solver.ode.y0
+    dy0 = deepcopy(y0)
+    solver.ode.F!(t0,y0,dy0)
+    step0 = Step(t0,deepcopy(y0),deepcopy(dy0))
+    step1 = Step(t0,deepcopy(y0),deepcopy(dy0))
+    solver_state = start(solver)
+    ytmp = deepcopy(y0)
+    return DenseState(step0, step1, t0-1, true, solver_state, ytmp, false)
+end
+
+# m3: I think it would be nice to factor out the dense-output and
+# root-finding into its own function.  That way it could be used also
+# independently of the dense-output iterator.  Also, it would make
+# this next function more compact.
+
+# pwl: I agree, but then the problem is that once you decouple them
+# you would lose the opprotunity to detect the roots with each step.
+function next(s::Solver{DenseStepper}, state::DenseState)
+
```

Comment:
But wouldn't you take one step with `next` which gets you to the next time, determined by the adaptive control. Then use that to create dense output and solve any equations.  So, you'd have an adaptive stepper `next` and then wrap that with a dense-output `next` which also does root finding.